pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -221,6 +201,26 @@ func (c *Controller) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 			Fn:           flow.TaskFn(botanist.WaitUntilEtcdForCopyDestroyed).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(botanist.IsRestorePhase()),
 			Dependencies: flow.NewTaskIDs(destroyETCDForCopy),
 		})
+		deployInfrastructure = g.Add(flow.Task{
+			Name:         "Deploying Shoot infrastructure",
+			Fn:           flow.TaskFn(botanist.DeployInfrastructure).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deploySecrets, deployCloudProviderSecret, deployReferencedResources, waitDestroyETCDForCopy),
```

Comment:
Just saw that by adding the dependency to waitDestroyETCDForCopy, we add a dependency to `deployBackupEntryInGarden` which shouldn't happen during normal reconciliation (all other steps that are taken during restoration would be skipped during reconciliation but this one still remains)